### PR TITLE
feat: INomadProtocol

### PR DESCRIPTION
**Pull request by anna-carroll**
## Motivation
We want a common interface through which mainnet forge scripts, mainnet fork testing, and local integration tests can be shared.

## Solution
Create interface INomadProtocol which mirrors the structure of the Nomad config

## PR Checklist

- [ ] Added Tests
- [ ] Updated Documentation
- [ ] Updated CHANGELOG.md for the appropriate package

### CODE REVIEW

**Comment by prestwich** [comment]
commit history is weird. have you been merging main instead of rebasing?
changes have been made to audited contracts, please revert

File: packages/contracts-ops/contracts/Config.sol [hunk 1]
```diff
@@ -72,6 +72,16 @@ abstract contract Config {
...
     function governanceRouter(string memory domain)
         public
+        override
+        onlyInitialized
+        returns (GovernanceRouter)
+    {
+        return GovernanceRouter(address(governanceRouterUpgrade(domain).proxy));
+    }
+
+    function governanceRouterUpgrade(string memory domain)
```
Comment (prestwich) [inline packages/contracts-ops/contracts/Config.sol:82]: nit: should be above `governanceRouter` which uses this function

File: packages/contracts-ops/contracts/Config.sol [hunk 2]
```diff
@@ -193,25 +240,60 @@ abstract contract Config {
...
+    function tokenRegistry(string memory domain)
+        public
+        override
+        onlyInitialized
+        returns (TokenRegistry)
+    {
+        return TokenRegistry(address(tokenRegistryUpgrade(domain).proxy));
+    }
+
     function tokenRegistryUpgrade(string memory domain)
```
Comment (prestwich) [inline packages/contracts-ops/contracts/Config.sol:260]: nit: should be above func that uses it

File: packages/contracts-ops/contracts/Config.sol [hunk 3]
```diff
@@ -193,25 +240,60 @@ abstract contract Config {
...
+        onlyInitialized
+        returns (AllowListNFTRecoveryAccountant)
+    {
+        return
+            AllowListNFTRecoveryAccountant(
+                address(accountantUpgrade(domain).proxy)
+            );
+    }
+
+    function accountantUpgrade(string memory domain)
```
Comment (prestwich) [inline packages/contracts-ops/contracts/Config.sol:282]: nit: "

File: packages/contracts-ops/contracts/Config.sol [hunk 4]
```diff
@@ -193,25 +240,60 @@ abstract contract Config {
...
+    {
+        return abi.decode(loadBridgeAttribute(domain, "accountant"), (Upgrade));
+    }
+
+    function ethHelper(string memory domain)
+        public
+        override
+        returns (ETHHelper)
+    {
+        return abi.decode(loadCoreAttribute(domain, "ethHelper"), (ETHHelper));
```
Comment (prestwich) [inline packages/contracts-ops/contracts/Config.sol:296]: this will error if ethHelper does not exist

**Review by prestwich — CHANGES_REQUESTED** [review]

File: packages/contracts-ops/contracts/Config.sol (hunk 3 above)
Comment (anna-carroll) [reply packages/contracts-ops/contracts/Config.sol]: https://github.com/nomad-xyz/monorepo/pull/484/commits/2d50e92c61b3ddbf5e85a9125d8204ef1ed675eb

File: packages/contracts-ops/contracts/Config.sol (hunk 2 above)
Comment (anna-carroll) [reply packages/contracts-ops/contracts/Config.sol]: https://github.com/nomad-xyz/monorepo/pull/484/commits/2d50e92c61b3ddbf5e85a9125d8204ef1ed675eb

File: packages/contracts-ops/contracts/Config.sol (hunk 1 above)
Comment (anna-carroll) [reply packages/contracts-ops/contracts/Config.sol:73]: https://github.com/nomad-xyz/monorepo/pull/484/commits/2d50e92c61b3ddbf5e85a9125d8204ef1ed675eb

File: packages/contracts-ops/contracts/Config.sol (hunk 4 above)
Comment (anna-carroll) [reply packages/contracts-ops/contracts/Config.sol]: https://github.com/nomad-xyz/monorepo/pull/484/commits/a857e10c6e8bc7d4bc29228593848b37747db6d4

**Comment by anna-carroll** [comment]
last commit is sort of drive-by but the fuzz tests were failing tests in CI for this PR

**Review by prestwich — APPROVED** [review]

**Pull request merged**